### DISCUSSION TITLE
Breakpoint on return

### DISCUSSION THREAD

**Discussion by Connell1** [dotnet/csharplang#4151] [Language Ideas] [answered]
While debugging in C#, I often find myself taking a method of the form:

`string DoSomething(object val)
{
    return SomeOtherMethod(val);
}`

and breaking it into multiple lines like:

`string DoSomething(object val)
{
    var result = SomeOtherMethod(val);
    return result;
}`

just so that I can set a breakpoint on the `return` and see the value. Would it be possible to make it so that on `Debug` builds, a breakpoint on a return statement would show the value that will be returned?

**Comment by YairHalberstadt** [comment, marked as answer]
This would be an issue for another repo. I'm not sure which one, but maybe Roslyn?

**Reply by Connell1** [reply]
Okay. I'll it asking it on the Roslyn repo. Thanks.

**Reply by 333fred** [reply]
And in fact, this feature already exists: the debug window should show "This method returned" after a method call.

**Comment by canton7** [comment]
(Opened as https://github.com/dotnet/roslyn/discussions/49466)
